atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -2922,4 +2954,13 @@ private <T> ListenableFuture<T> scopeToTransaction(ListenableFuture<T> transacti
                 },
                 MoreExecutors.directExecutor());
     }
+
+    @org.immutables.value.Value.Immutable
```

Comment:
Lets make this a package private immutable in its own class (and also get rid of the full prefix!). Should make the subsequent refactor a little easier too.